test/libsolidity/syntaxTests/bytecodeReferences/library_non_called.sol

```
@@ -7,5 +7,7 @@ library L2 {
 contract A {
     function f() public pure { type(L2).creationCode; }
```

Looks like `creationCode` and `runtimeCode` are producing broken IR on EOF. We need an assert against that in the codegen.

In the future it will also need an analysis check and an EOF-only copy of this test to test the error.

---

Same situation with:
- `gasleft()` in `497_gasleft.sol`.
- `selfdestruct()` in `address_payable_selfdestruct.sol`.
- `<address>.codehash` in `codehash.sol`.
- `<address>.code` in `address_members.sol`.
    - In this case we should also split the EOF version of the test into part that does not use `code` (it should pass) and the one for `code` (that should report an error and will only be possible after we add the analysis check).

---

Should `creationCode` and `runtimeCode` be unavailable for EOF? They brake code non observability rule, right? Assert is also needed, but it should be also changed in higher level to prevent usage of them in a contract code.

---

Done.